runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5216,6 +5217,117 @@ TR_J9ByteCodeIlGenerator::loadFlattenableInstanceWithHelper(int32_t cpIndex)
    push(helperCallNode);
    }
 
+static char * getTopLevelPrefixForFlattenedFields(TR_ResolvedJ9Method *owningMethod, int32_t cpIndex, int32_t &prefixLen, TR::Region &region)
+   {
+   int32_t len;
+   const char * fieldNameChars = owningMethod->fieldNameChars(cpIndex, len);
+   prefixLen = len + 1; // for '.'
+
+   char * newName = new (region) char[len+2];
+   strncpy(newName, fieldNameChars, len);
+
+   newName[len] = '.';
+   newName[len+1] = '\0';
+   return newName;
+   }
+
+void
+TR_J9ByteCodeIlGenerator::loadFlattenableInstance(int32_t cpIndex)
+   {
+   /* An example on what the tree with flattened fields looks like
+    *
+    * value NestedA {
+    *    int x;
+    *    int y;
+    *    }
+    * value NestedB {
+    *    NestedA a;
+    *    NestedA b;
+    *    }
+    * value ContainerC {
+    *    NestedB c;
+    *    NestedB d;
+    *    }
+    *
+    * method="ContainerC.getc()QNestedB;"
+    *
+    * /--- trees inserted ------------------------
+    * n5n      (  0)  NULLCHK on n3n [#32]
```

Comment:
why do we need a nullchk here? if the receiver is 'this' it would be guarnateed non-null... or is this some kind of more general transform and in this case we happen to have this and so we leave it to the optimizer to eliminate it later?

Reply:
I think I should check `!isNonNull()` before adding the `NULLCHK`. I'll update it next.